angular/projects/admin/src/app/admin/settings/settings.component.ts

```diff
@@ -84,6 +91,20 @@ export class SettingsComponent implements OnInit, OnDestroy {
     this.domainsFormArray.removeAt(index);
   }
 
+  dialogLanguage() {
+    const dialogRef = this.dialog.open(SettingsDialogManageLanguagesComponent, {
+      data: this.languages
+    });
+    dialogRef.afterClosed().subscribe(result => {
+      if (result !== undefined) {
+        if (result.status === 'submit') {
+          this.settingsForm.controls['defaultLanguage'].setValue('en');
```

Comment:
This is why the default language got reset to "en" after closing the dialog. 

Replace this with the following algorithm

 1. Default language remains unchanged if the current default language is still in the `result.languages` array
 2. Default language is set to `result.languages[0]` if the current default language is no longer in the `result.languages` array

Reply:
Solved on this commit https://github.com/oddbit/tanam/pull/51/commits/056bb5b2cfb37d0756e690536722e5cbd8ecff8f